src/index.js

```diff
@@ -579,31 +585,40 @@ export function useDropzone({
 
           const acceptedFiles = []
           const rejectedFiles = []
+          const rejectReasons = []
 
           files.forEach(file => {
-            if (fileAccepted(file, accept) && fileMatchSize(file, minSize, maxSize)) {
+            const [accepted, acceptError] = fileAccepted(file, accept)
+            const [sizeMatch, sizeError] = fileMatchSize(file, minSize, maxSize)
+            if (accepted && sizeMatch) {
               acceptedFiles.push(file)
             } else {
+              const error = acceptError || sizeError
```

Comment:
A file could have both type and size errors, I think you should account for that.